core/src/bank_forks.rs

```diff
@@ -252,14 +254,10 @@ impl BankForks {
                 .map_err(|_| BankForks::get_io_error("deserialize root error"));
             let status_cache: Result<StatusCacheRc, std::io::Error> = deserialize_from(&mut stream)
                 .map_err(|_| BankForks::get_io_error("deserialize bank status cache error"));
-            if bank_rc.is_none() {
-                let rc: Result<BankRc, std::io::Error> = deserialize_from(&mut stream)
-                    .map_err(|_| BankForks::get_io_error("deserialize bank accounts error"));
-                if rc.is_ok() {
-                    bank_rc = Some((rc.unwrap(), root.unwrap()));
-                }
+            if bank_root.is_none() && bank0.rc.update_from_stream(&mut stream).is_ok() {
```

Comment:
Why do we want the root on the first bank rather than the root on the latest bank?

Reply:
The bank root is restored and set from the latest snapshot as the snapshots are iterated in reverse sorted order.

Reply:
ah I see. Alternatively we could also reverse sort in the logic below using `vec.sort_by(|a, b| { a.name().cmp(b.name()).reverse() }` and just pass that in here instead of calling .rev